### PR TITLE
Implement MSC3890: Remotely silence local notifications

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR implements MSC3890. All that's needed from the homeserver side is to clean up some account data when a user's device is deleted.

Complement tests for this change are at https://github.com/matrix-org/complement/pull/576 (and will likely be the more complex part of this review).

Build failure in .deb packages will be fixed by https://github.com/matrix-org/synapse/pull/14774.

### CODE REVIEW

File: synapse/handlers/device.py [hunk 1]
```diff
@@ -499,7 +499,7 @@ async def delete_devices(self, user_id: str, device_ids: List[str]) -> None:
             else:
                 raise
 
-        # Delete access tokens and e2e keys for each device. Not optimised as it is not
+        # Delete data specific to this device. Not optimised as it is not
```
Comment (squahtx) [inline synapse/handlers/device.py:502]: really nitty wording preference:

We're deleting multiple devices here, so `each` makes more sense.
```suggestion
        # Delete data specific to each device. Not optimised as it is not
```

**Review by squahtx — APPROVED** [review]
lgtm!

File: synapse/handlers/device.py [hunk 2]
```diff
@@ -512,6 +512,14 @@ async def delete_devices(self, user_id: str, device_ids: List[str]) -> None:
                 user_id=user_id, device_id=device_id
             )
 
+            if self.hs.config.experimental.msc3890_enabled:
+                # Remove any local notification settings for this device in accordance
+                # with MSC3890.
+                await self.store.remove_account_data_for_user(
+                    user_id,
+                    f"org.matrix.msc3890.local_notification_settings.{device_id}",
+                )
```
Comment (anoadragon453) [inline synapse/handlers/device.py:521]: Hm, I've just noticed that we should use the [`AccountDataHandler.remove_account_data_for_user`](https://github.com/matrix-org/synapse/blob/c4456114e1a5471bb61cb45605e782263dc8233c/synapse/handlers/account_data.py#L215) method here instead, as that one will notify workers and Synapse modules that an update to account data has occurred.

Comment (anoadragon453) [reply synapse/handlers/device.py]: Fixed in 70d7cbf4f2796c84530c422a18d91e335fa8e766.

**Pull request merged**